### PR TITLE
[menu-bar] Migrate WebAuthenticationSession module to expo-modules and add electron support

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-11327

# How

- Migrate WebAuthenticationSession module to expo-modules 
- Add electron support

# Test Plan

https://github.com/expo/orbit/assets/11707729/6028fcb7-9fa7-4bc7-aa36-0a9e7800089e

https://github.com/expo/orbit/assets/11707729/8294367c-dd89-485d-a633-2dce992b68d5

### CODE REVIEW

File: apps/menu-bar/modules/web-authentication-session/ios/WebAuthenticationSessionModule.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Name("WebAuthenticationSession")
+
+    AsyncFunction("openAuthSessionAsync") { (urlString: String, promise: Promise) in
+      guard let url = URL(string: urlString) else {
+        promise.reject("INVALID_URL", "Invalid URL provided")
+        return
+      }
+
+      authSession = ASWebAuthenticationSession(url: url, callbackURLScheme: "expo-orbit") { callbackURL, error in
+        if let error = error {
```
Comment (alanjhughes) [inline apps/menu-bar/modules/web-authentication-session/ios/WebAuthenticationSessionModule.swift:18]:
```suggestion
        if let error {
```
If they have the same name we can omit the `= *`

File: apps/menu-bar/modules/web-authentication-session/ios/WebAuthenticationSessionModule.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          promise.reject("AUTH_SESSION_ERROR", error.localizedDescription)
+        } else {
+          if let callbackURL = callbackURL {
```
Comment (alanjhughes) [inline apps/menu-bar/modules/web-authentication-session/ios/WebAuthenticationSessionModule.swift:21]:
```suggestion
          if let callbackURL {
```

File: apps/menu-bar/modules/web-authentication-session/ios/WebAuthContextProvider.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import AuthenticationServices
+
+public class WebAuthContextProvider: NSObject, ObservableObject, ASWebAuthenticationPresentationContextProviding {
```
Comment (alanjhughes) [inline apps/menu-bar/modules/web-authentication-session/ios/WebAuthContextProvider.swift:3]: Do we need to conform to `ObservableObject` here? We're not using SwiftUI so I don't think it's necessary.

File: apps/menu-bar/modules/web-authentication-session/ios/WebAuthContextProvider.swift [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  public func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+    var anchor: ASPresentationAnchor?
+
+    DispatchQueue.main.sync {
+      anchor = NSApp.mainWindow!
+    }
+
+    return anchor!
```
Comment (alanjhughes) [inline apps/menu-bar/modules/web-authentication-session/ios/WebAuthContextProvider.swift:11]:
```suggestion
    var anchor: ASPresentationAnchor?

    DispatchQueue.main.sync {
      anchor = NSApp.mainWindow
    }

    return anchor ?? ASPresentationAnchor()
```
We can create our own instance and avoid force unwrapping. It should never happen that the mainWindow is nil but I think this is safer.

**Review by alanjhughes — APPROVED** [review]
LGTM ✅ Just some comments

File: apps/menu-bar/modules/web-authentication-session/ios/WebAuthContextProvider.swift (hunk 3 above)
Comment (gabrieldonadel) [reply apps/menu-bar/modules/web-authentication-session/ios/WebAuthContextProvider.swift]: You're right, no need to extend ObservableObject

**Pull request merged**